ci/pkl-pipelines/pay-deploy/concourse-runner.pkl

```diff
@@ -0,0 +1,251 @@
+              image_resource = new {
+                type = "registry-image"
+                source = new {
+                  ["repository"] = "concourse/oci-build-task"
+                }
+              }
+              inputs {
+                new {
+                  name = "concourse-runner-pr"
+                }
+              }
+              outputs {
+                new {
+                  name = "image"
+                }
+              }
+              run {
+                path = "build"
+              }
+            }
+          }
+          new TaskStep {
+            task = "build-concourse-runner-with-java-17"
+            privileged = true
+            output_mapping {
+              ["image"] = "concourse-runner-with-java-17-image"
+            }
+            params = new {
+              ["CONTEXT"] = "concourse-runner-pr/ci/docker/concourse-runner-with-java-17"
+              ["UNPACK_ROOTFS"] = "true"
+              ["DOCKER_CONFIG"] = "docker_creds"
+            }
+            config {
+              platform = "linux"
+              image_resource = new {
+                type = "registry-image"
+                source = new {
+                  ["repository"] = "concourse/oci-build-task"
+                }
+              }
+              inputs {
+                new {
+                  name = "concourse-runner-pr"
+                }
+              }
+              outputs {
+                new {
+                  name = "image"
+                }
+              }
+              run {
+                path = "build"
+              }
+            }
+          }
+        }
+      }
+      new InParallelStep {
+        in_parallel = new Listing<Step> {
+          testConcourseRunnerTask("concourse-runner-image", "adminusers-master")
+          (testConcourseRunnerTask("concourse-runner-with-java-17-image", "webhooks-main")){
+            task = "test-concourse-runner-with-java-17"
+          }
+        }
+      }
+      putPRTestStatus("concourse-runner-pr", "concourse-runner-tests", "success")
+    }
+    on_failure = putPRTestStatus("concourse-runner-pr", "concourse-runner-tests", "failure")
+    on_error = putPRTestStatus("concourse-runner-pr", "concourse-runner-tests", "failure")
+  }
+
+}
+
+local function withPath(path: String) = new Mixin{
+  source {
+    ["paths"] = new Listing<String> { path }
+  }
+}
+
+local function withPaths(paths: Listing<String>) = new Mixin{
+  source {
+    ["paths"] = paths
+  }
+}
+
+local function getStep(resource: String): GetStep = new {
+  get = resource
+}
+
+local function getStepWithTrigger(resource: String, _trigger: Boolean): GetStep = new {
+  get = resource
+  trigger = _trigger
+}
+
+local function getStepWithDepth(resource: String, depth: Int): GetStep = new {
+  get = resource
+  params = new {
+    ["depth"] = depth
+  }
+}
+
+local function putPRTestStatus(resourceName: String, test_name: String, status: String): Pipeline.PutStep = new {
```

Comment:
There is a mix of snake case and camel case for function variables. Shall we stick to camelCase for the variables?